src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
```

Comment:
PullProofAI

You can mark the `logger` property as readonly since it is initialized in the constructor and not reassigned.

```diff
-   private logger = new Logger();
+   private readonly logger = new Logger();
```



<!-- This is an auto-generated comment by PullProofAI -->